miden/src/main.rs

```diff
@@ -52,3 +54,22 @@ pub fn main() {
         println!("{}", error);
     }
 }
+
+// PROGRAM ERROR
+// ================================================================================================
+
+/// This is used to specify the error type returned from analyze.
+#[derive(Debug)]
+pub enum ProgramError {
+    AssemblyError(AssemblyError),
+    ExecutionError(ExecutionError),
+}
+
+impl fmt::Display for ProgramError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ProgramError::AssemblyError(e) => write!(f, "Assembly Error: {:?}", e),
+            ProgramError::ExecutionError(e) => write!(f, "Execution Error: {:?}", e),
+        }
+    }
+}
```

Comment:
Might be worthy to implement `std::error::Error` as well since many external crates depends on that for them to accept this a valid error type

Reply:
Agreed! Will add in the follow-up PR.